collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -145,6 +159,11 @@ private void scrapeBean(MBeanServerConnection beanConn, ObjectName mbeanName) {
                 logScrape(mbeanName, attr, "not readable");
                 continue;
             }
+
```

Comment:
I believe this would not work because we need to know which mBeans contain the excluded attributes. The `name` field within each `excludeBeanAttributeNames` record is "generic" since it can contain wildcards and other regex patterns. 

For example, suppose we have the following specified for `excludeBeanAttributeNames`:

```
excludeBeanAttributeNames:
 - name: "debezium.mysql:type=*,context=*,server=*"
   values:
    - TotalNumberOfEventsSeen
```
The `if (excludeBeanAttributeNames.contains(mbeanName) {` check would not work because `mbeanName` would contain the substituted regex patterns with actual values, like: `debezium.mysql:type=connector-metrics,context=streaming,server=dbserver1`. Please let me know if my understanding is incorrect.

Reply:
@dhoard

Reply:
@brandonstanleyappfolio

Given MBeans...

```
debezium.mysql:type=connector-metrics,context=streaming,server=dbserver1
debezium.mysql:type=connector-metrics,context=streaming,server=dbserver2
debezium.mysql:type=connector-metrics,context=streaming,server=dbserver3
```

And an `excludeBeanAttributeNames` pattern...

```
excludeBeanAttributeNames:
 - name: "debezium.mysql:type=*,context=*,server=server1"
   values:
    - TotalNumberOfEventsSeen
```

In the scenario above, we want to only exclude `TotalNumberOfEventsSeen` for the specific Object name that matches `debezium.mysql:type=*,context=*,server=server1`

While this is odd, it could happen for various reasons.

Can you add a test case with multiple MBeans and the above `excludeBeanAttributeNames` scenario?

Reply:
@dhoard Yep - added! Here is unit test: https://github.com/prometheus/jmx_exporter/blob/20977c1d11d8bb42e1c417aab4e57304d543b970/collector/src/test/java/io/prometheus/jmx/JmxCollectorTest.java#L373